test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+        # Rerun the job
+        print " qrerun job first time\n"
+
+        jids =  self.server.select(extend='T')
+        try:
+            self.server.rerunjob(jids)
+        except PbsRerunError:
+            pass
+
+        """
+        Verify that the accounting logs have R record with last known resource usage and run_count should be 2 for J1 and J2
+        """
+        time.sleep(6)
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 2}, jid1)
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 2}, jid2)
+
+        m = self.server.accounting_match(msg='.*R;' + jid1 + '.*Exit_status=-11.*.*resources_used.*.*resources_used.cput=[0-9]*:[0-9]*:[0-9]*.*.*run_count=1.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(msg='.*R;' + jid2 + '.*Exit_status=-11.*.*resources_used.*.*resources_used.cput=[0-9]*:[0-9]*:[0-9]*.*.*run_count=1.*', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+
+        #sleeping for 20 seconds
+        time.sleep(20)
+        # Rerun the job
+        print " qrerun job second time\n"
+
+        jids =  self.server.select(extend='T')
+
+        try:
+            self.server.rerunjob(jids)
+        except PbsRerunError:
+            pass
+
+        time.sleep(4)
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 3}, jid1)
+        self.server.expect(JOB, {ATTR_state: 'R', 'run_count': 3}, jid2)
+
+        m = self.server.accounting_match(msg='.*R;' + jid1 + '.*Exit_status=-11.*.*resources_used.*.*resources_used.cput=[0-9]*:[0-9]*:[0-9]*.*.*run_count=2.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(msg='.*R;' + jid2 + '.*Exit_status=-11.*.*resources_used.*.*resources_used.cput=[0-9]*:[0-9]*:[0-9]*.*.*run_count=2.*', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+
+        #sleeping for 10 seconds
+        time.sleep(10)
+        """
+        Delete all jobs
+        """
+        self.server.delete(jids)
+        print "deleted all jobs"
+        time.sleep(5)
+
+    def test_t7(self):
+        """
+        Scenario: Job is forcefully rerun.
+        Expected outcome: server should record last known resource usage in the R record.
+        """
+        dflt_q = self.server.default_queue
+
+        test = []
+        test += ['#PBS -N RequeueTest\n']
+        test += ['#PBS -l ncpus=1\n']
+        test += ['echo Starting test at `date`\n']
+        test += ['sleep 1000\n']
+
+        j1 = Job(TEST_USER)
+        j1.create_script(body=test)
+        j1.set_attributes({ATTR_r: 'y',ATTR_l + '.ncpus': 2})
+        jid1 = self.server.submit(j1)
+
+        j2 = Job(TEST_USER)
+        j2.create_script(body=test)
+        j2.set_attributes({ATTR_r: 'n',ATTR_l + '.ncpus': 2})
+        jid2 = self.server.submit(j2)
+
+        j3 = Job(TEST_USER)
+        j3.create_script(body=test)
+        j3.set_attributes({ATTR_r: 'y', ATTR_J: '1-4:1', ATTR_k: 'oe'})
```

Comment:
job arrays are rerunnable.  This can't be changed.  There's no need to say ATTR_r: 'y'